x-pack/solutions/observability/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -374,10 +376,18 @@ export const createMetricThresholdExecutor =
         const evaluationValues = getEvaluationValues<Evaluation>(alertResults, group);
         const thresholds = getThresholds<any>(criteria);
         const groups: Group[] = groupByMapping[group];
+        const flattenGrouping = groups
+          ? groups.reduce((acc, { field, value }) => {
+              return { ...acc, [field]: value };
+            }, {} as Record<string, string>)
+          : undefined;
+
+        const grouping = unflattenGrouping(flattenGrouping);
 
         const alertContext = {
           alertState: stateToAlertMessage[nextState],
           group,
+          grouping,
```

Comment:
@cesco-f in Metric and Log threshold rules, we have `groupByKeys` variable which is same as `grouping`, so I hadn't added it there. If you think we should add it, can you please also add it in list of variables in UI

Reply:
Done!